### PR TITLE
Add  getSpaceUser and getSpaceUsers endpoint in space api

### PR DESCRIPTION
**Pull request by vbeleuta**
We want to add user management endpoints to the api and document them.
This pull request adds the get `/space/{spaceId}/users` endpoints.

### CODE REVIEW

File: test/integration/user-integration.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+export default function userTests (t, space) {
+  t.test('Gets users', (t) => {
+    t.plan(2)
+    return space.getSpaceUsers()
+      .then((response) => {
+        t.ok(response.sys, 'sys')
+        t.ok(response.items, 'fields')
```
[the remark this answers is not in the document]
Comment (vbeleuta) [reply test/integration/user-integration.js]: yep, definitely; ill fix the comment and add more checks although for a new space might be items.length 0
however as we discussed I need to add the getSpaceUser so I will see if I can create space memberships before these tests (or move the usersTests to a testcase tied to a specific space)

File: lib/entities/user.js [hunk 2]
```diff
@@ -17,6 +17,17 @@ import cloneDeep from 'lodash/cloneDeep'
  * @prop {function(): Object} toPlainObject() - Returns this User as a plain JS object
  */
 
+/**
+ * @memberof User
+ * @typedef UserCollection
+ * @prop {number} total
+ * @prop {number} limit
+ * @prop {number} skip
+ * @prop {Object<{type: "Array"}>} sys
```
[the remark this answers is not in the document]
Comment (vbeleuta) [reply lib/entities/user.js]: yes the sys is outside the items array in the response, we have same typedef in UsageCollection

File: lib/entities/user.js [hunk 3]
```diff
@@ -17,6 +17,17 @@ import cloneDeep from 'lodash/cloneDeep'
  * @prop {function(): Object} toPlainObject() - Returns this User as a plain JS object
```
[the remark this answers is not in the document]
Comment (vbeleuta) [reply lib/entities/user.js]: I am not sure about this actually

File: lib/entities/user.js [hunk 4]
```diff
@@ -26,3 +37,14 @@ export function wrapUser (http, data) {
...
 }
+
+/**
+ * @private
+ * @param {Object} http - HTTP client instance
+ * @param {Object} data - Raw data collection
+ * @return {User.UserCollection} Normalized usage collection
+ */
+export function wrapUserCollection (http, data) {
+  const usage = toPlainObject(cloneDeep(data))
```
[the remark this answers is not in the document]
Comment (vbeleuta) [reply lib/entities/user.js]: oops, true! now I see its a copy paste i didnt fix, on it

File: lib/entities/user.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (phoebeschmidt) [reply lib/entities/user.js]: just check it and confirm bitte :) then this can be resolved

[the remark this answers is not in the document]
Comment (vbeleuta) [reply lib/entities/user.js]: the user object did not change and we do not manipulate it so we can keep this entity as is

**Pull request merged**